src/source/vector_tile_source.js

```diff
@@ -104,10 +106,15 @@ class VectorTileSource extends Evented implements Source {
     load() {
         this._loaded = false;
         this.fire(new Event('dataloading', {dataType: 'source'}));
-        this._tileJSONRequest = loadTileJSON(this._options, this.map._requestManager, (err, tileJSON) => {
+        const language = this.map._language;
+        const worldview = this.map._worldview;
+        this._tileJSONRequest = loadTileJSON(this._options, this.map._requestManager, language, worldview, (err, tileJSON) => {
             this._tileJSONRequest = null;
             this._loaded = true;
             if (err) {
+                if (language) console.warn(`Ensure that your requested language string is a valid BCP-47 code. Found: ${language}`);
+                if (worldview && worldview.length !== 2) console.warn(`Requested worldview strings must be a valid ISO alpha-2 code. Found: ${worldview}`);
+
```

Comment:
The validation of these fields should happen before sending the request. Check out [Intl.getCanonicalLocales](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/getCanonicalLocales)